pkg/controller/registry/resolver/operators.go

```diff
@@ -442,9 +440,24 @@ func PredicateForDependency(dependency *api.Dependency) (OperatorPredicate, erro
 	return p(dependency.Value)
 }
 
+func (o *Operator) DeprecatedPredicates() (predicates []OperatorPredicate, err error) {
+	predicates = make([]OperatorPredicate, 0)
+	for _, prop := range o.bundle.GetProperties() {
+		if prop.Type == string(opregistry.DeprecatedType) {
+			predicates = append(predicates, predicateForDeprecated(prop))
+		}
+	}
+	return
+}
+
+func predicateForDeprecated(prop *api.Property) OperatorPredicate {
+	return WithoutDeprecatedProperty()
+}
+
```

Comment:
You should be able to remove these, right? 

I don't see them used, and we don't need to build predicates dynamically based on the content of properties (OLM will just include the `WithoutDeprecated` predicate when filtering.